app/reducers/routes.js

```diff
@@ -13,7 +13,17 @@ export default function reducer(state = { scene: { notifications: [] } }, action
         scene: newScene,
       };
     case GAME_MEMO:
-      const newNotificaitons = state.scene.notifications.concat([action.payload]);
+      newNotifications = state.scene.notifications.concat([action.payload]);
+      newScene = Object.assign({}, state.scene, {
+        notifications: newNotifications
+      });
+      return {
+        ...state,
+        scene: newScene,
+      };
+    case DEQUEUE_GAME_MEMO:
+      newNotifications = state.scene.notifications.slice();
+      newNotifications.pop();
       newScene = Object.assign({}, state.scene, {
         notifications: newNotifications
```

Comment:
Optional refactor part 3:
remove line 25-26 and change 28 to:
`notifications: state.scene.notifications.slice(0, -1)`